arangod/RestServer/DatabaseFeature.cpp

```diff
@@ -619,15 +624,19 @@ int DatabaseFeature::createDatabase(TRI_voc_tick_t id, std::string const& name,
       }
     }
 
+    // we must actively copy the values to the builder that will provide the args-slice for the vocbase
     builder.openObject();
     builder.add("database", VPackValue(id));
     builder.add("id", VPackValue(std::to_string(id)));
     builder.add("name", VPackValue(name));
+
+    auto shardingReplicationFactorPair = getOneShardOptions(name, options);
```

Comment:
Why is there a method that returns replicationFactor and sharding info together, as a pair? For me this is kind of counter-intuitive.
Can't we have separate methods for the replicationFactor and for the sharding info?
Apart from that, the replicationFactor is not a one-shard-related option (i.e. it doesn't have anything to do with the one-shard feature).